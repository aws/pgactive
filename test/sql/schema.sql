--
-- pgactive tables' attributes must never change with schema changes.
-- Only new attributes can be appended and only if nullable.
--

select attrelid::regclass::text, attnum, attname, attisdropped, atttypid::regtype, attnotnull
from pg_attribute
where attrelid = ANY (ARRAY['pgactive.pgactive_nodes', 'pgactive.pgactive_connections', 'pgactive.pgactive_queued_drops', 'pgactive.pgactive_queued_commands', 'pgactive.pgactive_global_locks']::regclass[])
  and attnum >= 1
order by attrelid, attnum;
