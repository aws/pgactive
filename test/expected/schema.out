--
-- pgactive tables' attributes must never change with schema changes.
-- Only new attributes can be appended and only if nullable.
--
select attrelid::regclass::text, attnum, attname, attisdropped, atttypid::regtype, attnotnull
from pg_attribute
where attrelid = ANY (ARRAY['pgactive.pgactive_nodes', 'pgactive.pgactive_connections', 'pgactive.pgactive_queued_drops', 'pgactive.pgactive_queued_commands', 'pgactive.pgactive_global_locks']::regclass[])
  and attnum >= 1
order by attrelid, attnum;
        attrelid         | attnum |        attname        | attisdropped |         atttypid         | attnotnull 
-------------------------+--------+-----------------------+--------------+--------------------------+------------
 pgactive.pgactive_connections     |      1 | conn_sysid            | f            | text                     | t
 pgactive.pgactive_connections     |      2 | conn_timeline         | f            | oid                      | t
 pgactive.pgactive_connections     |      3 | conn_dboid            | f            | oid                      | t
 pgactive.pgactive_connections     |      4 | conn_origin_sysid     | f            | text                     | t
 pgactive.pgactive_connections     |      5 | conn_origin_timeline  | f            | oid                      | t
 pgactive.pgactive_connections     |      6 | conn_origin_dboid     | f            | oid                      | t
 pgactive.pgactive_connections     |      7 | conn_dsn              | f            | text                     | t
 pgactive.pgactive_connections     |      8 | conn_apply_delay      | f            | integer                  | f
 pgactive.pgactive_connections     |      9 | conn_replication_sets | f            | text[]                   | f
 pgactive.pgactive_global_locks    |      1 | locktype              | f            | text                     | t
 pgactive.pgactive_global_locks    |      2 | owning_sysid          | f            | text                     | t
 pgactive.pgactive_global_locks    |      3 | owning_timeline       | f            | oid                      | t
 pgactive.pgactive_global_locks    |      4 | owning_datid          | f            | oid                      | t
 pgactive.pgactive_global_locks    |      5 | owner_created_lock_at | f            | pg_lsn                   | t
 pgactive.pgactive_global_locks    |      6 | acquired_sysid        | f            | text                     | t
 pgactive.pgactive_global_locks    |      7 | acquired_timeline     | f            | oid                      | t
 pgactive.pgactive_global_locks    |      8 | acquired_datid        | f            | oid                      | t
 pgactive.pgactive_global_locks    |      9 | acquired_lock_at      | f            | pg_lsn                   | f
 pgactive.pgactive_global_locks    |     10 | state                 | f            | text                     | t
 pgactive.pgactive_nodes           |      1 | node_sysid            | f            | text                     | t
 pgactive.pgactive_nodes           |      2 | node_timeline         | f            | oid                      | t
 pgactive.pgactive_nodes           |      3 | node_dboid            | f            | oid                      | t
 pgactive.pgactive_nodes           |      4 | node_status           | f            | "char"                   | t
 pgactive.pgactive_nodes           |      5 | node_name             | f            | text                     | t
 pgactive.pgactive_nodes           |      6 | node_local_dsn        | f            | text                     | f
 pgactive.pgactive_nodes           |      7 | node_init_from_dsn    | f            | text                     | f
 pgactive.pgactive_nodes           |      8 | node_read_only        | f            | boolean                  | f
 pgactive.pgactive_nodes           |      9 | node_seq_id           | f            | smallint                 | f
 pgactive.pgactive_queued_commands |      1 | lsn                   | f            | pg_lsn                   | t
 pgactive.pgactive_queued_commands |      2 | queued_at             | f            | timestamp with time zone | t
 pgactive.pgactive_queued_commands |      3 | perpetrator           | f            | text                     | t
 pgactive.pgactive_queued_commands |      4 | command_tag           | f            | text                     | t
 pgactive.pgactive_queued_commands |      5 | command               | f            | text                     | t
 pgactive.pgactive_queued_commands |      6 | search_path           | f            | text                     | f
 pgactive.pgactive_queued_drops    |      1 | lsn                   | f            | pg_lsn                   | t
 pgactive.pgactive_queued_drops    |      2 | queued_at             | f            | timestamp with time zone | t
 pgactive.pgactive_queued_drops    |      3 | dropped_objects       | f            | pgactive.dropped_object[]     | t
(37 rows)

